src/vision/status_quo/alan_builds_a_task_scheduler.md

```diff
@@ -0,0 +1,704 @@
+# 😱 Status quo stories: Alan builds a task scheduler
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+A core component of [`DistriData`][distridata], called `TaskScheduler`, is in charge of (1) receiving client requests via an HTTP server, (2) serializing them in a task queue, (3) relaying each task to the applier, and (4) returning the result back to the client.
+
+`TaskScheduler` was originally implemented in Go. New to Rust, [Alan][] believes Rust could provide the same quality of service but with less memory. Then decides to reimplement `TaskScheduler` in Rust, without knowing the challenges ahead.
+
+### **Implementing request ID generator**
+
+Alan first transliterates request ID generator code, originally written in Go:
+
+```go
+import "sync/atomic"
+
+type Generator interface {
+	next() uint64
+}
+
+type generator struct {
+	prefix uint64
+	suffix uint64
+}
+
+func (gen *generator) next() uint64 {
+	suffix := atomic.SwapUint64(&gen.suffix, gen.suffix+1)
+	id := gen.prefix | (suffix & (math.MaxUint64 >> 16))
+	return id
+}
+```
+
+Alan learns Rust trait as the closest concept to Go interface but is now torn between [std::sync::atomic](https://doc.rust-lang.org/std/sync/atomic) and [crossbeam::atomic::AtomicCell](https://docs.rs/crossbeam). Reading multiple articles about how great crossbeam is and for its thread-safety promises, Alan chooses crossbeam (see ["crates better than std (from Reddit)"](https://www.reddit.com/r/rust/comments/hat5bt/what_are_your_favorite_better_than_std_crates/)):
+
+```rust
+use crossbeam::atomic::AtomicCell;
+
+pub struct Generator {
+    prefix: u64,
+    suffix: AtomicCell<u64>,
+}
+
+impl Generator {
+    pub fn new(...) -> Self {
+        ...
+    }
+
+    pub fn next(&self) -> u64 {
+        let suffix = self.suffix.fetch_add(1);
+        let id = self.prefix | (suffix & (u64::MAX >> 16));
+        id
+    }
+}
+```
+
+Accustomed to an opinionated way of doing concurrency in Go, Alan loses confidence in Rust async support, as he sees fragmented but specialized solutions in Rust async ecosystem.
+
+### **Implementing event notifier**
+
+Alan then implements notifier to propagate request apply progress with scheduler and low-level state machine. In Go, it can be simply implemented as below:
```

Comment:
I'm not sure I understand what this sentence is saying. Can you rephrase this? I attempted adding what appeared to be some missing comas and "the"s, but I'm still not sure. 

```suggestion
Alan then implements the notifier to propagate the request and apply the progress with the scheduler and low-level state machine. In Go, it can be simply implemented as below:
```